### ISSUE TITLE
TenFourFox on Intel - "OS is not defined"

### ISSUE THREAD

**Issue by kencu** [classilla/tenfourfox#496]
[Please excuse me if this is off-topic for TenFourFox on PPC]

I'm entering this text now from a debug build of TenFourFox built and running on MacOSX 10.6.8. Enough of it works to be able to do this. Lots of kudos to @rmottola who together with @classilla did all the heavy lifting, and little bits thrown in from elsewhere. Also many thanks to the folks at MacPorts who make a lot of this possible by keeping the compiler chain healthy on these older systems, especially @ryandesign.

One of the early observations is that there is a disconnect between the menu selections and actions they should be taking (they don't do anything), and entering text into the search bar or URL bar doesn't send the browser off.

There seems to be a consistent error popping up relating to "OS is not defined" that I thought might be able to draw some insight:

```
./firefox
1523421353702	addons.xpi	ERROR	Failed to process extension changes at startup: [Exception... "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIXPCComponents_Utils.import]"  nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)"  location: "JS frame :: resource://gre/modules/osfile/osfile_shared_allthreads.jsm :: <TOP_LEVEL> :: line 92"  data: no] Stack trace: resource://gre/modules/osfile/osfile_shared_allthreads.jsm:92 < resource://gre/modules/osfile/osfile_async_front.jsm:28 < resource://gre/modules/osfile.jsm:11 < resource://gre/modules/DeferredSave.jsm:11 < XPCU_moduleLambda()@resource://gre/modules/XPCOMUtils.jsm:273 < XPCU_defineLazyGetter/<.get()@resource://gre/modules/XPCOMUtils.jsm:198 < this.XPIDatabase.saveChanges()@resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:461 < this.XPIDatabase.removeAddonMetadata()@resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:1322 < this.XPIDatabase.updateAddonMetadata()@resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:1302 < updateMetadata()@resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:1793 < processFileChanges()@resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:1946 < this.XPIProvider.checkForChanges()@resource://gre/modules/addons/XPIProvider.jsm:3647 < this.XPIProvider.startup()@resource://gre/modules/addons/XPIProvider.jsm:2655 < callProvider()@resource://gre/modules/AddonManager.jsm:227 < _startProvider()@resource://gre/modules/AddonManager.jsm:833 < AddonManagerInternal.startup()@resource://gre/modules/AddonManager.jsm:1016 < this.AddonManagerPrivate.startup()@resource://gre/modules/AddonManager.jsm:2783 < amManager.prototype.observe()@addonManager.js:58
2018-04-10 21:35:53.947 firefox[88887:903] *** __NSAutoreleaseNoPool(): Object 0x503450 of class NSCFDictionary autoreleased with no pool in place - just leaking
2018-04-10 21:35:53.949 firefox[88887:903] *** __NSAutoreleaseNoPool(): Object 0x26e960 of class NSCFData autoreleased with no pool in place - just leaking
*************************
A coding exception was thrown and uncaught in a Task.

Full message: ReferenceError: OS is not defined
Full stack: AddonDatabase.openConnection/this.connectionPromise<@resource://gre/modules/addons/AddonRepository.jsm:1605:1
TaskImpl_run@resource://gre/modules/Task.jsm:315:40
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
Task_spawn@resource://gre/modules/Task.jsm:164:12
AddonDatabase.openConnection@resource://gre/modules/addons/AddonRepository.jsm:1580:31
AddonDatabase.retrieveStoredData@resource://gre/modules/addons/AddonRepository.jsm:1753:12
this.AddonRepository.getCachedAddonByID<@resource://gre/modules/addons/AddonRepository.jsm:584:7
TaskImpl_run@resource://gre/modules/Task.jsm:315:40
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
getRepositoryAddon@resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:152:3
this.XPIDatabase.getAddon/<@resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:1107:9
Handler.prototype.process@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:933:23
this.PromiseWalker.walkerLoop@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:812:7
this.PromiseWalker.scheduleWalkerLoop/<@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:746:1

*************************
*************************
A coding exception was thrown and uncaught in a Task.

Full message: ReferenceError: OS is not defined
Full stack: SRCH_SVC__asyncReadCacheFile/<@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3201:1
TaskImpl_run@resource://gre/modules/Task.jsm:331:41
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
Task_spawn@resource://gre/modules/Task.jsm:164:12
SRCH_SVC__asyncReadCacheFile@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3187:1
SRCH_SVC__asyncInit/<@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:2707:21
TaskImpl_run@resource://gre/modules/Task.jsm:331:41
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
Task_spawn@resource://gre/modules/Task.jsm:164:12
SRCH_SVC__asyncInit@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:2699:1
task@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3750:17
TaskImpl_run@resource://gre/modules/Task.jsm:331:41
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
Task_spawn@resource://gre/modules/Task.jsm:164:12
SRCH_SVC_init@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3747:7
searchbar_XBL_Constructor@chrome://browser/content/search/search.xml:76:1

*************************
*************************
A coding exception was thrown and uncaught in a Task.

Full message: ReferenceError: OS is not defined
Full stack: SRCH_SVC__asyncReadCacheFile/<@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3201:1
TaskImpl_run@resource://gre/modules/Task.jsm:331:41
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
Task_spawn@resource://gre/modules/Task.jsm:164:12
SRCH_SVC__asyncReadCacheFile@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3187:1
SRCH_SVC__asyncInit/<@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:2707:21
TaskImpl_run@resource://gre/modules/Task.jsm:331:41
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
Task_spawn@resource://gre/modules/Task.jsm:164:12
SRCH_SVC__asyncInit@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:2699:1
task@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3750:17
TaskImpl_run@resource://gre/modules/Task.jsm:331:41
TaskImpl@resource://gre/modules/Task.jsm:276:3
createAsyncFunction/asyncFunction@resource://gre/modules/Task.jsm:250:14
Task_spawn@resource://gre/modules/Task.jsm:164:12
SRCH_SVC_init@file:///Users/Shared/tenfourfox/obj-ff-dbg/dist/TenFourFox.app/Contents/Resources/components/nsSearchService.js:3747:7
searchbar_XBL_Constructor@chrome://browser/content/search/search.xml:76:1

```

**Comment by kencu** [comment]
Here's a more complete run of the entire error log from a 20 minute run of TFF-Intel. [TFF.Intel.runerrors.txt](https://github.com/classilla/tenfourfox/files/1897350/TFF.Intel.runerrors.txt)

**Comment by rmottola** [comment]
I don't know if that is an error or actually a failure to report it properly.
I am more concerned with the many undefined:

Full message: ReferenceError: gAsyncDBWrapperPromised is not defined

your log is full of scary things, but not the same I get.

Startup TTF/Intel and just type something in the url bar and hit enter. See what you get in the console.

**Comment by classilla** [comment]
Seems like `osfile.jsm` isn't loading for some reason.

**Comment by kencu** [comment]
Looks like the fix for this is to re-enable ctypes. That required a little x86 fix to libffi courtesy of MacPorts.

Now it's all working very nicely, menus, URL bar, etc. Just needs some reasonable testing to find out what we don't know we don't know!

**Comment by classilla** [comment]
Can these fixes be added to #494? I'd rather take everything at once.

**Comment by rmottola** [comment]
They are in? Can you pull now? I'd like to work on a separate of 
patches then - not to mix with these.
